### PR TITLE
Adding relative path to websocket uri

### PR DESCRIPTION
**Pull request by twhtanghk**
Updated to master branch again for travis error reported. Adding relative path to websocket uri for single page application deploying to run on different context path. Also, update readme with steps for running novnc in docker container. Please review and merge.

### CODE REVIEW

**Comment by CendioOssman** [comment]
Could you add some automatic tests that shows correct behaviour for this?

**Comment by twhtanghk** [comment]
Simplified the change in rfb.js to verify the relpath again. I found difficulty to setup test environment and create fakesocket with relative uri in test.websock.js under docker container.

**Comment by CendioOssman** [comment]
Are you able to run the existing tests? You should be able to run the tests by doing:

```
$ npm install
$ npm test
```

And then opening the specified test URL in a browser.

**Comment by twhtanghk** [comment]
It is fine to run npm install and test. Browser loads html and js properly, but web socket url for test environment is not relative to the deployed url in similar situation for rjb.js. Any hints to change web socket url for test environment.

**Comment by CendioOssman** [comment]
I'll have a look and see if I can help.

**Comment by DirectXMan12** [comment]
I'd also like to get the dockerfile additions in a separate PR, so we can debate the merits of that separately.

File: core/rfb.js [hunk 1]
```diff
@@ -392,7 +392,10 @@ RFB.prototype = {
             uri = this._encrypt ? 'wss' : 'ws';
         }
 
-        uri += '://' + this._rfb_host + ':' + this._rfb_port + '/' + this._rfb_path;
+        var relpath = location.pathname.split('/');
```
Comment (DirectXMan12) [inline core/rfb.js]: `core/rfb.js` *should not* depend on the current location -- that's something that should be handled in the UI.  If you want a relative path, do it in the UI code that sets the RFB path.

Comment (CendioOssman) [reply core/rfb.js]: So your vote is to reject this PR?

Comment (DirectXMan12) [reply core/rfb.js]: yeah, if we do this, I'd rather see it done in `app/ui.js`.

Comment (twhtanghk) [reply core/rfb.js]: ui.js keeps the websocket path in localstorage and reload it from settings saved earlier. Just lookup for any other way to override the settings by url (e.g. https://mppsrc.ogcio.hksarg/twhtang/novnc/vnc.html?path=/context/novnc/websockify). It simplifies the integration for other web app to invoke novnc by opening new tab for different vnc console (e.g. window.open 'https://mppsrc.ogcio.hksarg/twhtang/novnc/vnc.html', '_blank').

Comment (twhtanghk) [reply core/rfb.js]: path overriding is already implemented in WebUitl.getQueryVar via query param. That is, no source code change is required. Thanks.

**Pull request closed without merging**